File: Go/ExampleTests/production/shoppingcart.go
package domain

type ShoppingCart struct {
	Items OrderItemCollection
	Shipping Dollar
	Discount Dollar
	TotalPrice Dollar
	ShippingAddress Address
	BillingAddress Address
}

func NewShoppingCart(items OrderItemCollection, shipping Dollar, discount Dollar, totalprice Dollar, shippingaddress Address, billingaddress Address) *ShoppingCart {
	return &ShoppingCart{Items: items, Shipping: shipping, Discount: discount, TotalPrice: totalprice, ShippingAddress: shippingaddress, BillingAddress: billingaddress}
}

File: Swift/ExampleTests/production/ShoppingCart.swift
public struct ShoppingCart {
    public let items: OrderItemCollection
    public let shipping: Dollar
    public let discount: Dollar
    public let totalPrice: Dollar
    public let shippingAddress: Address
    public let billingAddress: Address

    public init(items: OrderItemCollection, shipping: Dollar, discount: Dollar, totalPrice: Dollar, shippingAddress: Address, billingAddress: Address) {
        self.items = items
        self.shipping = shipping
        self.discount = discount
        self.totalPrice = totalPrice
        self.shippingAddress = shippingAddress
        self.billingAddress = billingAddress
    }
}
